cgra/CGRAKingMeshRTL.py

```diff
@@ -149,11 +149,23 @@ def construct( s, DataType, PredicateType, CtrlType, width, height,
         s.tile[i].to_mem_waddr.rdy //= 0
         s.tile[i].to_mem_wdata.rdy //= 0
 
-  # Line trace
-  def line_trace( s ):
-    # str = "||".join([ x.element.line_trace() for x in s.tile ])
-    # str += " :: [" + s.data_mem.line_trace() + "]"
-    res = "||\n".join([ (("[tile"+str(i)+"]: ") + x.line_trace() + x.ctrl_mem.line_trace())
-                              for (i,x) in enumerate(s.tile) ])
-    res += "\n :: [" + s.data_mem.line_trace() + "]    \n"
+  # verbose trace if verbosity > 0
+  def verbose_trace(s, verbosity=1):
+    res = ''
+    for (i, x) in enumerate(s.tile):
+      res += "# [tile" + str(i) + "]: " + x.line_trace(verbosity=verbosity) + x.ctrl_mem.line_trace(verbosity=verbosity) + '\n'
+    res += f"\ndata_mem: {s.data_mem.line_trace(verbosity=verbosity)}"
+    res += "------\n\n"
     return res
+
+  # Line trace
+  def line_trace(s, verbosity=0):
+    if verbosity==0:
+      # str = "||".join([ x.element.line_trace() for x in s.tile ])

```

Comment:
Space before and after all operators, e.g., `verbosity=0` -> ` verbosity = 0 `, `verbosity==0` -> `verbosity == 0`. Plz all this to all the changes.

Reply:
Got it.

Reply:
done